File: backend/app/entities/document.go
package entities

type Document struct {
	Base
	Title       string      `json:"title" gorm:"not null"`
	Description string      `json:"description" gorm:"type:text"`
	Content     string      `json:"content" gorm:"type:text"`
	FileName    string      `json:"file_name"`
	FileType    string      `json:"file_type"`
	FileSize    int64       `json:"file_size"`
	ChunkCount  int         `json:"chunk_count" gorm:"default:0"`
	Status      string      `json:"status" gorm:"default:'pending'"` // pending, processing, ready, failed
	Metadata    SingleJSONB `json:"metadata" gorm:"type:jsonb"`
	Chatbots    []Chatbot   `json:"chatbots,omitempty" gorm:"many2many:chatbot_documents;"`
}

func (Document) TableName() string {
	return "documents"
}

File: frontend/src/types/document.ts
export interface IDocument {
  id: number
  title: string
  description: string
  content?: string
  file_name: string
  file_type: string
  file_size: number
  chunk_count: number
  status: string // pending, processing, ready, failed
  metadata?: Record<string, any>
  created_at?: string
  updated_at?: string
}
